apps/demos-e2e/src/integration/app.spec.ts

```diff
@@ -1,13 +1,57 @@
-import { getGreeting } from '../support/app.po';
+import * as Devtools from '../support/devtools-protocol-tasks.utils';
 
-describe('demos', () => {
-  beforeEach(() => cy.visit('/'));
+const WAIT_DURATION = 1000;
 
-  it('should display welcome message', () => {
-    // Custom command example, see `../support/commands.ts` file
-    cy.login('my-email@something.com', 'myPassword');
+describe('RxLet vs Push demo', () => {
+  beforeEach(() => cy.task('resetCRI').visit('/performance/rx-let-vs-push/list-toggle'));
 
-    // Function helper example, see `../support/app.po.ts` file
-    getGreeting().contains('Welcome to demos!');
+  describe('test and generate Profiler reports', () => {
+    it('should check performance of push pipe', () => {
+      cy.get('button').contains('Open Manual test for Push').click();
+
+      Devtools.startProfiler();
+      testOpeningAndClosingToggleThreeTimes();
+      Devtools.stopProfiler('push-pipe');
+    });
+
+    it('should check performance of let directive', () => {
+      cy.get('button').contains('Open Manual test for Let').click();
+
+      Devtools.startProfiler();
+      testOpeningAndClosingToggleThreeTimes();
+      Devtools.stopProfiler('let-directive');
+    });
+  });
+
+  describe('test and generate Tracing reports', () => {
+    it('should check performance of push pipe', () => {
+      cy.get('button').contains('Open Manual test for Push').click();
+
+      Devtools.startTracing();
+      testOpeningAndClosingToggleThreeTimes();
+      Devtools.stopTracing('push-pipe');
+    });
+
+    it('should check performance of let directive', () => {
+      cy.get('button').contains('Open Manual test for Let').click();
+
+      Devtools.startTracing();
+      testOpeningAndClosingToggleThreeTimes();
+      Devtools.stopTracing('let-directive');
+    });
   });
 });
+
+function openAndCloseToggle() {
+  cy.get('button').contains('Toggle').click();
+  cy.get('rxa-rendering-work').contains('0');
+  cy.wait(WAIT_DURATION);
+  cy.get('button').contains('Toggle').click();
+  cy.wait(WAIT_DURATION);
+}
+
+function testOpeningAndClosingToggleThreeTimes() {
+  openAndCloseToggle();
+  openAndCloseToggle();
+  openAndCloseToggle();
+}
```

Comment:
should we move that in a po file?

Reply:
It's good to set up some basic good practices for writing future tests, event if it is just an example test. So I moved it to a PO file as suggested :)